Range-v3 failed to build due to lots of errors with MSVC on windows

Environment:
Windows Server 2016 + VS2019 + Range-v3 master branch latest srouce code.

Range-v3 failed to build due to lots of errors with MSVC on windows . It can be reproduced on master latest revision 6dd1cb6. Could you help have a look about this issue? Thanks in advance!

Note: 
1. Add the /std:c++17 option to build successfully

Steps to reproduce the behavior:

1. git clone https://github.com/ericniebler/range-v3 D:\Range-v3\src
2. Open a VS 2019 x64 command prompt as admin and browse to D:\Range-v3
3.  mkdir build_x64 && pushd build_x64
4. cmake -G "Visual Studio 16 2019" -A x64 -DCMAKE_SYSTEM_VERSION=10.0.18362.0 -DCMAKE_BUILD_TYPE=Release
5. pop
6. pushd build_x64
7. msbuild /m /p:Platform=x64 /p:Configuration=Release Range-v3.sln /t:Rebuild 
log_x64_build.log

Actual result:
103>F:\gitP\ericniebler\range-v3\include\range/v3/detail/config.hpp(222,1): fatal error C1189: #error:  range-v3 requires Visual Studio 2019 with the /std:c++17 (or /std:c++latest) and /permissive- options. [F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.min.vcxproj]
2020-07-28T05:45:24.1089797Z    181>ClCompile:
2020-07-28T05:45:24.1090161Z          unique_copy.cpp
2020-07-28T05:45:24.1090507Z    165>ClCompile:
2020-07-28T05:45:24.1090979Z          set_union1.cpp
2020-07-28T05:45:24.1091302Z    180>ClCompile:
2020-07-28T05:45:24.1091641Z          upper_bound.cpp
2020-07-28T05:45:24.1092213Z    103>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.min.vcxproj" (Rebuild target(s)) -- FAILED.
2020-07-28T05:45:24.1093002Z     69>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.min.vcxproj.metaproj" (Rebuild target(s)) -- FAILED.
2020-07-28T05:45:24.1093575Z    192>CustomBuild:
2020-07-28T05:45:24.1094114Z          Building Custom Rule F:/gitP/ericniebler/range-v3/example/CMakeLists.txt
2020-07-28T05:45:24.1094576Z    186>ClCompile:
2020-07-28T05:45:24.1095868Z          C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.26.28801\bin\HostX64\x64\CL.exe /c /I"F:\gitP\ericniebler\range-v3\include" /nologo /W3 /WX /diagnostics:column /O2 /Ob2 /D WIN32 /D _WINDOWS /D NDEBUG /D RANGES_CXX_ALIGNED_NEW=0 /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /EHsc /MD /GS /fp:precise /permissive- /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++14 /Fo"config.dir\Release\\" /Fd"config.dir\Release\vc142.pdb" /Gd /TP /errorReport:queue "F:\gitP\ericniebler\range-v3\test\config.cpp"
2020-07-28T05:45:24.1097097Z          config.cpp
2020-07-28T05:45:24.1097540Z    179>ClCompile:
2020-07-28T05:45:24.1098079Z          any_all_none_of.cpp
2020-07-28T05:45:24.1150497Z    184>ClCompile:
2020-07-28T05:45:24.1285529Z          C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.26.28801\bin\HostX64\x64\CL.exe /c /I"F:\gitP\ericniebler\range-v3\include" /nologo /W3 /WX /diagnostics:column /O2 /Ob2 /D WIN32 /D _WINDOWS /D NDEBUG /D RANGES_CXX_ALIGNED_NEW=0 /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /EHsc /MD /GS /fp:precise /permissive- /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++14 /Fo"comprehension_conversion.dir\Release\\" /Fd"comprehension_conversion.dir\Release\vc142.pdb" /Gd /TP /errorReport:queue "F:\gitP\ericniebler\range-v3\example\comprehension_conversion.cpp"
2020-07-28T05:45:24.1287849Z     94>F:\gitP\ericniebler\range-v3\include\range/v3/detail/config.hpp(222,1): fatal error C1189: #error:  range-v3 requires Visual Studio 2019 with the /std:c++17 (or /std:c++latest) and /permissive- options. [F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.partition_point.vcxproj]
2020-07-28T05:45:24.1378757Z      2>Project "F:\gitP\ericniebler\range-v3\build_amd64\ALL_BUILD.vcxproj.metaproj" (2) is building "F:\gitP\ericniebler\range-v3\build_amd64\test\iterator\iter.basic_iterator.vcxproj.metaproj" (196) on node 1 (Rebuild target(s)).
2020-07-28T05:45:24.1380037Z    196>Project "F:\gitP\ericniebler\range-v3\build_amd64\test\iterator\iter.basic_iterator.vcxproj.metaproj" (196) is building "F:\gitP\ericniebler\range-v3\build_amd64\test\iterator\iter.basic_iterator.vcxproj" (202) on node 16 (Rebuild target(s)).
2020-07-28T05:45:24.1545395Z    202>CoreClean:
2020-07-28T05:45:24.1546101Z          Creating directory "iter.basic_iterator.dir\Release\".
2020-07-28T05:45:24.1546677Z     94>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.partition_point.vcxproj" (Rebuild target(s)) -- FAILED.
2020-07-28T05:45:24.1547735Z    107>F:\gitP\ericniebler\range-v3\include\range/v3/detail/config.hpp(222,1): fatal error C1189: #error:  range-v3 requires Visual Studio 2019 with the /std:c++17 (or /std:c++latest) and /permissive- options. [F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.max_element.vcxproj]
2020-07-28T05:45:24.1570742Z     83>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.partition_point.vcxproj.metaproj" (Rebuild target(s)) -- FAILED.
2020-07-28T05:45:24.1599492Z    107>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.max_element.vcxproj" (Rebuild target(s)) -- FAILED.
2020-07-28T05:45:24.1627072Z     67>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.max_element.vcxproj.metaproj" (Rebuild target(s)) -- FAILED.
2020-07-28T05:45:24.1628238Z      2>Project "F:\gitP\ericniebler\range-v3\build_amd64\ALL_BUILD.vcxproj.metaproj" (2) is building "F:\gitP\ericniebler\range-v3\build_amd64\test\iterator\iter.reverse_iterator.vcxproj.metaproj" (200) on node 1 (Rebuild target(s)).
2020-07-28T05:45:24.1629498Z    200>Project "F:\gitP\ericniebler\range-v3\build_amd64\test\iterator\iter.reverse_iterator.vcxproj.metaproj" (200) is building "F:\gitP\ericniebler\range-v3\build_amd64\test\iterator\iter.reverse_iterator.vcxproj" (204) on node 11 (Rebuild target(s)).
2020-07-28T05:45:24.1630296Z    204>CoreClean:
2020-07-28T05:45:24.1630721Z          Creating directory "iter.reverse_iterator.dir\Release\".
2020-07-28T05:45:24.1631190Z    202>PrepareForBuild:
2020-07-28T05:45:24.1631698Z          Creating directory "F:\gitP\ericniebler\range-v3\build_amd64\test\iterator\Release\".
2020-07-28T05:45:24.1632305Z          Creating directory "iter.basic_iterator.dir\Release\iter.bas.913BE758.tlog\".
2020-07-28T05:45:24.1633133Z      2>Project "F:\gitP\ericniebler\range-v3\build_amd64\ALL_BUILD.vcxproj.metaproj" (2) is building "F:\gitP\ericniebler\range-v3\build_amd64\test\multiple.vcxproj.metaproj" (201) on node 1 (Rebuild target(s)).
2020-07-28T05:45:24.1634236Z    201>Project "F:\gitP\ericniebler\range-v3\build_amd64\test\multiple.vcxproj.metaproj" (201) is building "F:\gitP\ericniebler\range-v3\build_amd64\test\multiple.vcxproj" (208) on node 12 (Rebuild target(s)).
2020-07-28T05:45:24.1634948Z    208>CoreClean:
2020-07-28T05:45:24.1635341Z          Creating directory "multiple.dir\Release\".
2020-07-28T05:45:24.1636367Z    202>InitializeBuildStatus:
2020-07-28T05:45:24.1637216Z          Creating "iter.basic_iterator.dir\Release\iter.bas.913BE758.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
2020-07-28T05:45:24.1637757Z    184>ClCompile:
2020-07-28T05:45:24.1638109Z          comprehension_conversion.cpp
2020-07-28T05:45:24.1639287Z     99>F:\gitP\ericniebler\range-v3\include\range/v3/detail/config.hpp(222,1): fatal error C1189: #error:  range-v3 requires Visual Studio 2019 with the /std:c++17 (or /std:c++latest) and /permissive- options. [F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.partial_sort_copy.vcxproj]
2020-07-28T05:45:24.1640119Z    208>PrepareForBuild:
2020-07-28T05:45:24.1640452Z          Creating directory "multiple.dir\Release\multiple.tlog\".
2020-07-28T05:45:24.1640836Z        InitializeBuildStatus:
2020-07-28T05:45:24.1723585Z          Creating "multiple.dir\Release\multiple.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
2020-07-28T05:45:24.1857089Z    192>ClCompile:
2020-07-28T05:45:24.1859826Z          C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.26.28801\bin\HostX64\x64\CL.exe /c /I"F:\gitP\ericniebler\range-v3\include" /nologo /W3 /WX /diagnostics:column /O2 /Ob2 /D WIN32 /D _WINDOWS /D NDEBUG /D RANGES_CXX_ALIGNED_NEW=0 /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /EHsc /MD /GS /fp:precise /permissive- /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++14 /Fo"count.dir\Release\\" /Fd"count.dir\Release\vc142.pdb" /Gd /TP /errorReport:queue "F:\gitP\ericniebler\range-v3\example\count.cpp"
2020-07-28T05:45:24.1967975Z    202>CustomBuild:
2020-07-28T05:45:24.1968629Z          Building Custom Rule F:/gitP/ericniebler/range-v3/test/iterator/CMakeLists.txt
2020-07-28T05:45:24.2466514Z     99>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.partial_sort_copy.vcxproj" (Rebuild target(s)) -- FAILED.
2020-07-28T05:45:24.2467833Z    186>F:\gitP\ericniebler\range-v3\include\range/v3/detail/config.hpp(222,1): fatal error C1189: #error:  range-v3 requires Visual Studio 2019 with the /std:c++17 (or /std:c++latest) and /permissive- options. [F:\gitP\ericniebler\range-v3\build_amd64\test\config.vcxproj]
2020-07-28T05:45:24.2740990Z    208>CustomBuild:
2020-07-28T05:45:24.2741843Z          Building Custom Rule F:/gitP/ericniebler/range-v3/test/CMakeLists.txt
2020-07-28T05:45:24.2825436Z     80>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\algorithm\alg.partial_sort_copy.vcxproj.metaproj" (Rebuild target(s)) -- FAILED.
2020-07-28T05:45:24.2826294Z    186>Done Building Project "F:\gitP\ericniebler\range-v3\build_amd64\test\config.vcxproj" (Rebuild target(s)) -- FAILED.

--- a/cmake/ranges_flags.cmake
+++ b/cmake/ranges_flags.cmake
@@ -60,7 +60,11 @@ if(CMAKE_CXX_STANDARD)
     cxx_standard_denormalize(${CMAKE_CXX_STANDARD} RANGES_CXX_STD)
   endif()
 elseif("x${RANGES_CXX_STD}" STREQUAL "xdefault")
-  set(RANGES_CXX_STD 14)
+  if (RANGES_CXX_COMPILER_CLANGCL OR RANGES_CXX_COMPILER_MSVC)
+    set(RANGES_CXX_STD 17)
+  else()
+    set(RANGES_CXX_STD 14)
+  endif()
 endif()
 
 # All compilation flags

